src/common/libjob/Makefile.am

```diff
@@ -68,3 +71,7 @@ test_sign_none_t_LDADD = $(test_ldadd) $(LIBDL)
 test_specutil_t_SOURCES = test/specutil.c
 test_specutil_t_CPPFLAGS = $(test_cppflags)
 test_specutil_t_LDADD = $(test_ldadd) $(LIBDL)
+
+test_spec_t_SOURCES = test/spec.c
+test_spec_t_CPPFLAGS = $(test_cppflags)
+test_spec_t_LDADD = $(test_ldadd) $(LIBDL)
```

Comment:
Suggestion: rename the source files to jobspec1.c and jobspec1.h, and the unit test to jobspec1.

It may get awkward later if user are including "spec.h" and we introduce a jobspec2 interface, for example.